.github/workflows/ci.yaml

```diff
@@ -18,6 +18,12 @@ on:
 
 jobs:
   build_and_test:
+    env:
+      # The special value "local" tells Go to use the bundled Go version rather
+      # than trying to fetch one according to a `toolchain` value in `go.mod`.
+      # This ensures that we're really running the Go version in the CI matrix
+      # rather than one that the Go command has upgraded to automatically.
+      GOTOOLCHAIN: local
```

Comment:
@bgentry I think this is the magic invocation needed to stop CI from autoupgrading its toolchain. A value of "local" tells Go to ignore a `go.mod` `toolchain` directive and use the version in the installed Go command.

See this section of the docs:

https://go.dev/doc/toolchain#select

> When GOTOOLCHAIN is set to local, the go command always runs the bundled Go toolchain.